packages/mds-compliance-service/repository/index.ts

```diff
@@ -146,12 +149,84 @@ class ComplianceSnapshotReadWriteRepository extends ReadWriteRepository {
     }
   }
 
+  public getComplianceViolationPeriods = async (
```

Comment:
Given this is a giant super complicated SQL blob, probably ought to have a comment explaining what it's doing